stdlib/public/core/StringStorageBridge.swift

```diff
@@ -335,6 +373,464 @@ extension __SharedStringStorage {
     // reference will make the instance non-unique.
     return self
   }
+  
+  @objc(copy)
+  final internal func copy() -> AnyObject {
+    return self
+  }
+  
+  @objc(mutableCopyWithZone:)
+  final internal func mutableCopy(with zone: _SwiftNSZone?) -> AnyObject {
+    return _SwiftNSMutableString(self.asString)
+  }
+  
+  @objc(mutableCopy)
+  final internal func mutableCopy() -> AnyObject {
+    return _SwiftNSMutableString(self.asString)
+  }
+  
+}
+
+@usableFromInline
+@objc internal final class _SwiftNSMutableString :
+  _SwiftNativeNSMutableString, _AbstractStringStorage {
+  
+  // Swift String always wants to have valid contents, so for cases where
+  // mutation of the contents makes them invalid (e.g. deleting half of a
+  // surrogate pair) we have an alternate backing store that's just an
+  // Array of UTF16, which we largely leave up to NSString to interpret
+  private enum Contents {
+    case native(String)
+    case utf16([UTF16.CodeUnit])
+    
+    var asString: String {
```

Comment:
Add explicit version control on everything here

Reply:
I'm not following, sorry. Why would this need versioning?